staging/src/k8s.io/cloud-provider/controllers/service/controller.go

```diff
@@ -735,8 +735,9 @@ func (c *Controller) nodeSyncService(svc *v1.Service, oldNodes, newNodes []*v1.N
 	}
 	newNodes = filterWithPredicates(newNodes, getNodePredicatesForService(svc)...)
 	oldNodes = filterWithPredicates(oldNodes, getNodePredicatesForService(svc)...)
-	if nodeNames(newNodes).Equal(nodeNames(oldNodes)) {
-		return retSuccess
+
+	if nodeNames(newNodes).Equal(nodeNames(oldNodes)) && providerIDsEqual(oldNodes, newNodes) {
+		return retNeedRetry
```

Comment:
Why did this return code change?  I feel like I am missing something...

Reply:
yyeah, sames nodes and with the same providersids looks like means `retSuccess`

Reply:
do we want to be paranoid and and check the node.uid too ? https://github.com/kubernetes/cloud-provider-gcp/issues/589 , VMs can be registered with the same name and providerID but be different 🤷

Reply:
WDYT?

Reply:
Wouldn't keying the nodes by their uid instead of their names, address that? So: `nodeUID(newNodes).Equal(nodeUID(oldNodes))`.

Reply:
I like that (with a big comment why :)